web/yo/app/scripts/controllers/drugs.js

```diff
@@ -31,24 +61,40 @@ angular.module('oncokbApp')
 
             $scope.saveDrugName = function (newDrugName, drug) {
                 if (hasSameName(newDrugName, drug.uuid)) {
-                    modalError("Sorry", "Same name exists.", false, drug);
+                    modalError("Sorry", "Same name exists.", true, false, drug.uuid);
                 } else {
                     if (!newDrugName)
                         newDrugName = drug.drugName;
                     firebaseConnector.set('Drugs/' + drug.uuid + '/drugName', newDrugName);
                 }
             };
+
+            $scope.removeDrug = function (drug) {
+                firebaseConnector.ref('Map').once('value', function (snapshot) {
```

Comment:
In here, you don't need to refetch data from firebase. You can check if the uuid exists in _.keys($scope.drugMap) directly when you use `on` in previous comment, .